src/tutorialGBIF_occurrence_maps.Rmd

```diff
@@ -0,0 +1,175 @@
+```
+
+### Get occurrences from GBIF
+Get occurrence data from GBIF via function `occ_data` from `RGBIF` package:
+```{r}
+occurrences <- map(taxonKeys, 
+                   function(x) rgbif::occ_data(
+                     taxonKey = x, 
+                     country =  paste(unlist(countries), 
+                                      collapse = ";"), 
+                     year = paste(unlist(year_range), collapse = ","),
+                     limit = 200000)$data)
+occurrences <- set_names(occurrences, nm = species)
+```
+
+###Overview 
+A first overview of species distribution:
+```{r}
+longitude_max <- max(unlist(map(occurrences, 
+                                function(spec) 
+                                  max(spec$decimalLongitude, na.rm = TRUE))))
+longitude_min <- min(unlist(map(occurrences, 
+                                function(spec) 
+                                  min(spec$decimalLongitude, na.rm = TRUE))))
+
+latitude_max <- max(unlist(map(occurrences, 
+                                function(spec) 
+                                  max(spec$decimalLatitude, na.rm = TRUE))))
+latitude_min <- min(unlist(map(occurrences, 
+                                function(spec) 
+                                  min(spec$decimalLatitude, na.rm = TRUE))))
+purrr::map(names(occurrences), function(spec) 
+  qmplot(x = decimalLongitude,
+         y = decimalLatitude, 
+         data = occurrences[[spec]][1:1000,], 
+         maptype = "toner-lite", 
+         xlim = c(longitude_min, longitude_max), 
+         ylim = c(latitude_min,latitude_max),
+         main = spec,
+         color = I("blue")))
+```
+
+### Monthly occurrence distribution
+For some species can be interesting to see the monthly occurrence distribution:
+```{r}
+months <- c(1:12)
+occurrences_per_month <- map(months, function(m) map(occurrences, function(x) x %>% filter(month == m)))
+counts_per_months <- map(species, 
+                         function(i) unlist(
+                           map(occurrences_per_month, 
+                               function(x) 
+                                 nrow(x[[i]]))))
+counts_per_months <- set_names(counts_per_months, nm = species)
+counts_per_months_df <- tidyr::gather(data.frame(counts_per_months), species)
+counts_per_months_df$month <- rep.int(1:12, length(species))
+ ggplot(counts_per_months_df, aes(fill = species, y = value, x = month)) +
+   geom_bar(position="dodge", stat="identity") +
+   scale_x_continuous(breaks=seq(1,12,1), 
+                     labels=c("Jan", "Feb", "Mar", "Apr", "May", "Jun",
+                              "Jul", "Aug", "Sept", "Oct", "Nov",
+                              "Dec"))
+```
+
+### Occurrence geographical distribution based on UTM-10 squares
+Import Belgium UTM-10 square polygon data:
+```{r}
+eu_10grid <-  readOGR("../data/input/EUgrid10.geojson")
+wgs_84 <- CRS("+init=epsg:4326") # CRS used by GBIF
+eu_10grid <- spTransform(eu_10grid, wgs_84)
```

Comment:
This seems needed. Without `spTransform`:
`eu_10grid@proj4string` gives
`CRS arguments:
 +proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0 `. Afterwards:
`CRS arguments:
 +init=epsg:4326 +proj=longlat +datum=WGS84 +no_defs +ellps=WGS84
+towgs84=0,0,0 `
So, without `spTransform`, applying function `over` on the `eu10_grid` and the points, returns the following error:
`Error: identicalCRS(x, y) is not TRUE`

Reply:
ah, maybe the Transform is not needed, but just adding the definition of the WGS84 to the `eu_10grid` variable.  Just as a check, define the CRS of `eu_10grid` as WGS84. I do think this will be something like:

```
CRS(eu_10grid) <- wgs_84
```

Reply:
`CRS` is not a valid function. However, I found the needed function, `proj4string`, in a nice ![overview](https://www.nceas.ucsb.edu/~frazier/RSpatialGuides/OverviewCoordinateReferenceSystems.pdf "Overview CRSs").
`proj4string(eu_10grid) <- wgs_84`
Obvoiusly I get the warning:
`A new CRS was assigned to an object with an existing CRS:
+proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0
without reprojecting.
For reprojection, use function spTransform`
In our case `towgs84=0,0,0` that means that `eu_10grid` is already in wgs84 :+1: